receiver/elasticsearchreceiver/scraper.go

```diff
@@ -0,0 +1,223 @@
+
+			r.metricsBuilder.RecordElasticsearchNodeThreadPoolTasksQueuedDataPoint(r.now, tpInfo.QueuedTasks, tpName)
+
+			r.metricsBuilder.RecordElasticsearchNodeThreadPoolTasksFinishedDataPoint(r.now, tpInfo.CompletedTasks, tpName, metadata.AttributeTaskState.Completed)
+			r.metricsBuilder.RecordElasticsearchNodeThreadPoolTasksFinishedDataPoint(r.now, tpInfo.RejectedTasks, tpName, metadata.AttributeTaskState.Rejected)
+		}
+
+		r.metricsBuilder.RecordElasticsearchNodeDocumentsDataPoint(r.now, info.Indices.DocumentStats.ActiveCount, metadata.AttributeDocumentState.Active)
+		r.metricsBuilder.RecordElasticsearchNodeDocumentsDataPoint(r.now, info.Indices.DocumentStats.DeletedCount, metadata.AttributeDocumentState.Deleted)
+
+		r.metricsBuilder.RecordElasticsearchNodeOpenFilesDataPoint(r.now, info.ProcessStats.OpenFileDescriptorsCount)
+
+		r.metricsBuilder.RecordJvmClassesLoadedDataPoint(r.now, info.JVMInfo.ClassInfo.CurrentLoadedCount)
+
+		r.metricsBuilder.RecordJvmGcCollectionsCountDataPoint(r.now, info.JVMInfo.JVMGCInfo.Collectors.Young.CollectionCount, "young")
+		r.metricsBuilder.RecordJvmGcCollectionsCountDataPoint(r.now, info.JVMInfo.JVMGCInfo.Collectors.Old.CollectionCount, "old")
+
+		r.metricsBuilder.RecordJvmGcCollectionsElapsedDataPoint(r.now, info.JVMInfo.JVMGCInfo.Collectors.Young.CollectionTimeInMillis, "young")
+		r.metricsBuilder.RecordJvmGcCollectionsElapsedDataPoint(r.now, info.JVMInfo.JVMGCInfo.Collectors.Old.CollectionTimeInMillis, "old")
+
+		r.metricsBuilder.RecordJvmMemoryHeapMaxDataPoint(r.now, info.JVMInfo.JVMMemoryInfo.MaxHeapInBy)
+		r.metricsBuilder.RecordJvmMemoryHeapUsedDataPoint(r.now, info.JVMInfo.JVMMemoryInfo.HeapUsedInBy)
+		r.metricsBuilder.RecordJvmMemoryHeapCommittedDataPoint(r.now, info.JVMInfo.JVMMemoryInfo.HeapCommittedInBy)
+
+		r.metricsBuilder.RecordJvmMemoryNonheapUsedDataPoint(r.now, info.JVMInfo.JVMMemoryInfo.NonHeapUsedInBy)
+		r.metricsBuilder.RecordJvmMemoryNonheapCommittedDataPoint(r.now, info.JVMInfo.JVMMemoryInfo.NonHeapComittedInBy)
+
+		r.metricsBuilder.RecordJvmMemoryPoolUsedDataPoint(r.now, info.JVMInfo.JVMMemoryInfo.MemoryPools.Young.MemUsedBy, "young")
+		r.metricsBuilder.RecordJvmMemoryPoolUsedDataPoint(r.now, info.JVMInfo.JVMMemoryInfo.MemoryPools.Survivor.MemUsedBy, "survivor")
+		r.metricsBuilder.RecordJvmMemoryPoolUsedDataPoint(r.now, info.JVMInfo.JVMMemoryInfo.MemoryPools.Old.MemUsedBy, "old")
+
+		r.metricsBuilder.RecordJvmMemoryPoolMaxDataPoint(r.now, info.JVMInfo.JVMMemoryInfo.MemoryPools.Young.MemMaxBy, "young")
+		r.metricsBuilder.RecordJvmMemoryPoolMaxDataPoint(r.now, info.JVMInfo.JVMMemoryInfo.MemoryPools.Survivor.MemMaxBy, "survivor")
+		r.metricsBuilder.RecordJvmMemoryPoolMaxDataPoint(r.now, info.JVMInfo.JVMMemoryInfo.MemoryPools.Old.MemMaxBy, "old")
+
+		r.metricsBuilder.RecordJvmThreadsCountDataPoint(r.now, info.JVMInfo.JVMThreadInfo.Count)
+
+		r.metricsBuilder.EmitNodeMetrics(ilms.Metrics())
```

Comment:
Great to see so much done in such few lines. Thanks @dmitryax!

Reply:
Seriously, the new MetricBuilder stuff is really well done!